pkg/ebpf/telemetry/errors_telemetry.go

```diff
@@ -50,15 +50,31 @@ type EBPFTelemetry struct {
 	probeKeys    map[string]uint64
 }
 
-// NewEBPFTelemetry initializes a new EBPFTelemetry object
-func NewEBPFTelemetry() *EBPFTelemetry {
-	if supported, _ := ebpfTelemetrySupported(); !supported {
-		return nil
-	}
-	return &EBPFTelemetry{
+// A singleton instance of the ebpf telemetry struct. Used by the collector and the ebpf managers (via ErrorsTelemetryModifier).
+var errorsTelemetry *EBPFTelemetry
+
+// newEBPFTelemetry initializes a new EBPFTelemetry object
+func newEBPFTelemetry() *EBPFTelemetry {
```

Comment:
If `newEBPFTelemetry` is called for a second time we would just create another instance, should we control that case?

Reply:
initially i called this method `getEBPFTelemetry` and returned an existing instance if already initialized. The issue are with tests, when we run tests that run different build modes, they are failing on the second attempt, since the struct is already initialized and contains (already) invalid pointers to the ebpf maps